modules/tensor_mechanics/src/materials/ADLAROMANCEStressUpdateBase.C

```diff
@@ -169,350 +226,319 @@ ADLAROMANCEStressUpdateBase::initQpStatefulProperties()
-                   ": previously calculated scalar (",
-                   MetaPhysicL::raw_value(scalar),
-                   ") is too high resulting in a negative trial stress (",
-                   MetaPhysicL::raw_value(trial_stress_mpa),
-                   "). Cutting timestep.");
-
-  Real effective_strain_old;
-  if (_creep_strain_old_forcing_function)
-    effective_strain_old = _creep_strain_old_forcing_function->value(_t, _q_point[_qp]);
-  else
-    effective_strain_old =
-        std::sqrt(_creep_strain_old[_qp].doubleContraction(_creep_strain_old[_qp]) / 1.5);
-
-  ADReal rom_effective_strain = 0.0;
-  ADReal derivative_rom_effective_strain = 0.0;
-
-  computeROMStrainRate(_dt,
-                       _mobile_old,
-                       _immobile_old,
-                       trial_stress_mpa,
-                       effective_strain_old,
-                       _temperature[_qp],
-                       _environmental[_qp],
-                       _mobile_dislocation_increment,
-                       _immobile_dislocation_increment,
-                       rom_effective_strain,
-                       derivative_rom_effective_strain);
+  ADReal rom_effective_strain_inc = 0.0;
+  ADReal drom_effective_strain_inc_dstress = 0.0;
+
+  // Check window of new trial stress
+  _extrapolation[_qp] =
+      _non_stress_extrapolation * checkInputWindow(_input_values[_stress_input_idx],
+                                                   _input_limits[_stress_input_idx],
+                                                   _window_failure[_stress_input_idx]);
+
+  // Run ROM if all values are within windows.
+  if (_extrapolation[_qp])
+  {
+    rom_effective_strain_inc = computeROMStrainRate(_strain_output_idx);
+    drom_effective_strain_inc_dstress = computeROMStrainRate(_strain_output_idx, true);
+
+    // apply extrapolation
+    rom_effective_strain_inc *= _extrapolation[_qp];
+    const ADReal dextrapolation_dstress =
+        _non_stress_extrapolation * checkInputWindow(_input_values[_stress_input_idx],
+                                                     _input_limits[_stress_input_idx],
+                                                     _window_failure[_stress_input_idx],
+                                                     true);
+    drom_effective_strain_inc_dstress = drom_effective_strain_inc_dstress * _extrapolation[_qp] +
+                                        rom_effective_strain_inc * dextrapolation_dstress;
+  }
 
   if (_verbose)
   {
-    Moose::out << "Verbose information from " << _name << ": \n";
-    Moose::out << "  dt: " << _dt << "\n";
-    Moose::out << "  old mobile disl: " << _mobile_old << "\n";
-    Moose::out << "  old immobile disl: " << _immobile_old << "\n";
-    Moose::out << "  initial stress (MPa): "
+    ADReal environmental = 0.0;
+    if (_environmental)
+      environmental = (*_environmental)[_qp];
+    Moose::err << "Verbose information from " << _name << ": \n";
```

Comment:
Seems odd to send this to the error stream (since it's not an error).

Reply:
Yeah, this is to ensure that it is actually output when running in parallel. If it is just Moose::out and not on the main mpi process, it will be buried.